### PR TITLE
Integrate Forms Playground (Part 1)

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Related to #3126 
Depends on https://github.com/camunda/form-playground/pull/1

---- 

**This includes**
* [x] Replace the existing form editor  
* [x] Add toggle to the status bar
* [x] Error Panel reduces Viewport Size instead of Overlapping
* [x] Persist toggle states across sessions
* [x] Add keyboard shortcut to toggle between modes (`CTRL/CMD + P`, as we already have with BPMN/DMN Properties Panel and DMN Overview)

Cf. to https://github.com/camunda/camunda-modeler/issues/3126 to check out what is _not included_ (tracking etc.)

**Artifacts**
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/3126-integrate-forms-playground/camunda-modeler-3126-integrate-forms-playground-linux-x64.tar.gz
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/3126-integrate-forms-playground/camunda-modeler-3126-integrate-forms-playground-mac.dmg
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/3126-integrate-forms-playground/camunda-modeler-3126-integrate-forms-playground-mac.zip
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/3126-integrate-forms-playground/camunda-modeler-3126-integrate-forms-playground-win-ia32.zip
* https://camunda-modeler-on-demand.s3.eu-central-1.amazonaws.com/3126-integrate-forms-playground/camunda-modeler-3126-integrate-forms-playground-win-x64.zip

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
I'd like to put this one under review before we integrate the resizable panels (cf. https://github.com/camunda/form-playground/issues/3). The current scope of the PR is already quite big, and I don't want to have this blocked for the time being. Feel free to disagree 🙂 

I'm adding @YanaSegal and @christian-konrad as reviewers. I added artifacts to try it out in the issue description.

**Comment by nikku** [comment]
@Skaiir Please leave your review on this one.

**Comment by christian-konrad** [comment]
It works well! In case we can fix the form output/submit data thing independently, lgtm

File: client/src/app/tabs/form/FormPreviewToggle.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    onCollapsePreview,
+    onOpenPreview,
+    previewOpen
+  } = props;
+
+  return <Fill slot="status-bar__app" group="1_form-preview">
+    <div className={ classnames(css.FormPreviewToggle) }>
+      <button
+        className={ classnames('btn', { 'btn--active': !previewOpen }) }
+        title="Open design"
```
Comment (christian-konrad) [inline client/src/app/tabs/form/FormPreviewToggle.js:35]: I assume "Open design mode" is better when it comes to a11y (respectively "Open validation mode")

File: client/src/app/tabs/form/getFormWindowMenu.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+export function getFormWindowMenu(state) {
+  return [
+    ...getPreviewEntries(state)
+  ];
+}
+
+function getPreviewEntries({ previewOpen }) {
+  return [ {
+    label: previewOpen ? 'Open design' : 'Open validation',
```
Comment (christian-konrad) [inline client/src/app/tabs/form/getFormWindowMenu.js:19]: Same here ("Open design mode", "Open validation mode")

**Review by christian-konrad — APPROVED** [review]
🚀

File: client/src/app/tabs/form/FormPreviewToggle.js (hunk 1 above)
Comment (pinussilvestrus) [reply client/src/app/tabs/form/FormPreviewToggle.js]: Updated via b5225cc56ca87456047c39d9986abc83f5931ea1 👍

**Review by Skaiir — COMMENTED** [review]
I'm not approving because I don't want to auto-merge accidentally in case we want to get https://github.com/bpmn-io/form-js/pull/364 through before, which will require a re-release. 

Tested manually and went over all the code, everything looks as it should IMO 👍

**Comment by pinussilvestrus** [comment]
I fixed that; there was some outdated flexbox styling in the FormEditor. Interestingly this worked in the build-on-demand artifact, it probably had an older form-js version included.

image

**Comment by pinussilvestrus** [comment]
I agree, let's wait for https://github.com/bpmn-io/form-js/pull/364 to be released and integrated to merge this one.

**Comment by Skaiir** [comment]
Just found this with tab navigation:

electron_DpPGvjGYIq

Looks like a tabindex and visibility issue.

To replicate, select the top form element and tab ahead a bunch of times.

**Comment by Skaiir** [comment]
I think a quick fix would be to make these bottom ones tabIndex = -1 in design mode.

**Comment by Skaiir** [comment]
Or even better, straight up not rendering this in design mode:

image

**Comment by christian-konrad** [comment]
> Or even better, straight up not rendering this in design mode

This makes sense to me, does it come with caveats?

**Comment by pinussilvestrus** [comment]
Should be fixed via https://github.com/camunda/form-playground/pull/18. Just needs to create a patch release of the playground.

**Review by Skaiir — APPROVED** [review]
LGTM 🎉

**Comment by pinussilvestrus** [comment]
@YanaSegal @RomanKostka I just merged this now to get it out of the door; it's hanging for quite a while now. Feel free to add design suggestions later on 👍

**Comment by YanaSegal** [comment]
@pinussilvestrus FYI, @RomanKostka will takeover from here.

**Pull request merged**